hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java

```diff
@@ -182,7 +182,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       long correctedSpace = omKeyInfo.getReplicatedSize() -
           allocatedLocationInfoList.size() * scmBlockSize * factor;
       // Subtract the size of blocks to be overwritten.
-      if (keyToDelete != null) {
+      if (keyToDelete != null && !omBucketInfo.getIsVersionEnabled()) {
```

Comment:
Nit: Might be cleaner to combine the identical if conditions here and line 158

Reply:
Good point. Let's combine the identical.